trace_replay/block_cache_tracer.h

```diff
@@ -197,6 +197,8 @@ class BlockCacheTracer {
   // Stop writing block cache accesses to the trace_writer.
   void EndTrace();
 
+  bool is_tracing_enabled() const { return writer_.load(); }
```

Comment:
Do you need to use `load(std::memory_order_relaxed)` here?

Reply:
I think we should stick to the default (sequential consistency) here. The reason is that all threads should start tracing once the block cache tracer is enabled.

Reply:
I don't think you need such strictness. I believe you just need to depend on the atomicity of writer_ rather than full-scale synchronization between threads. std::memory_order_relaxed feels like it should sufficient for your use case.

Reply:
Synchronized load is quite expensive. It's better to avoid. I agree with @sagar0 that a clean cut is not necessary in this case. Even if a CS load is made, the ordering is still hard to reason among threads.

Reply:
I updated to use memory_order_relaxed.